sdks/java/core/src/main/java/org/apache/beam/sdk/util/TimerInternals.java

```diff
@@ -63,12 +63,11 @@
   void setTimer(StateNamespace namespace, String timerId, Instant target, TimeDomain timeDomain);
 
   /**
-   * Writes out a timer to be fired when the watermark reaches the given timestamp, automatically
-   * generating an id for it from the provided {@link TimerData}.
+   * Writes out a timer to be fired when the watermark reaches the timestamp contained in the
```

Comment:
"Writes ..." seems a bit odd. Maybe just "Sets the timer described by {@code timerKey}"

Also, I think timerKey should be renamed to timerData?